rust/numaflow-core/src/config/components.rs

```diff
@@ -37,7 +38,7 @@ pub(crate) mod source {
         Generator(GeneratorConfig),
         UserDefined(UserDefinedConfig),
         Pulsar(PulsarSourceConfig),
-        Serving(serving::Settings),
+        Serving(Arc<serving::Settings>),
```

Comment:
why `Arc`, is it for easy cloning? how come this is different from the rest? please document the reason?